backend/routes/rankings.js

```diff
@@ -1,52 +1,81 @@
-            where: { UserId: user.id },
-            include: [{ model: Stock }]
-        });
-
-        const stocksYouFollow = follows.length === 0 ? [] : follows.map((follow) => follow.Stock);
-
-        const stocksInDatabase = await Stock.findAll();
+        const [stocksInDatabase, follows, likes, dislikes] = await Promise.all([
+            Stock.findAll(),
+            Follow.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Like.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Dislike.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            })
+        ]);
 
         if (stocksInDatabase.length < 10) {
             res.status(422).json({
                 error: "You need at least 10 stocks in the database to access ranking."
             });
         }
 
-        const randomStocksNeeded = 10 - stocksYouFollow.length;
+        const stocksYouFollow = follows.map((follow) => follow.Stock);
+        const stocksYouLike = likes.map((like) => like.Stock);
+        const stocksYouDislike = dislikes.map((dislike) => dislike.Stock);
 
-        const stocksRanking = [...stocksYouFollow];
+        const stocksRanking = [];
 
-        for (let i = 0; i < randomStocksNeeded; i++) {
-            let potentialStock;
-            let alreadyInRanking;
+        stocksInDatabase.forEach((currentStock) => {
+            let points = 0;
 
-            do {
-                const randomIndex = Math.floor(
-                    Math.random() * stocksInDatabase.length
-                );
-                potentialStock = stocksInDatabase[randomIndex];
+            const selectedStockYouFollow = stocksYouFollow.find((stock) => stock.ticker === currentStock.ticker);
 
-                alreadyInRanking = stocksRanking.some(
-                    (stock) => stock.name === potentialStock.name
-                );
-            } while (alreadyInRanking);
+            if (selectedStockYouFollow != null) {
+                points += FOLLOW_POINTS;
+            }
 
-            stocksRanking.push(potentialStock);
-        }
+            const selectedStockYouLike = stocksYouLike.find((stock) => stock.ticker === currentStock.ticker);
+
+            if (selectedStockYouLike != null) {
+                points += LIKE_POINTS;
+            }
+
+            const selectedStockYouDislike = stocksYouDislike.find((stock) => stock.ticker === currentStock.ticker);
+
+            if (selectedStockYouDislike != null) {
+                points += DISLIKE_POINTS;
+            }
+
+            const { ticker, name, logo } = currentStock.dataValues;
+            const stockWithPoints = { ticker, name, logo, points }
+
+            stocksRanking.push(stockWithPoints);
+        });
+
+        stocksRanking.sort(compareStocksByPoints);
```

Comment:
one nice-to-have is to display score (or just # of likes and dislikes) on the UI in the rankings list